sympy/polys/rings.py

```diff
@@ -2531,15 +2531,169 @@ def compose(f, x, a=None):
 
         return poly
 
+    def coeff_wrt(self, x, deg):
+        """
+        Coefficient of ``self`` with respect to ``x**deg``.
+
+        Treating ``self`` as a univariate polynomial in ``x`` this finds the
+        coefficient of ``x**deg`` as a polynomial in the other generators.
+
+        Parameters
+        ==========
+
+        x : generator or generator index
+            The generator or generator index to compute the expression for.
+        deg : int
+            The degree of the monomial to compute the expression for.
+
+        Returns
+        =======
+
+        PolyElement
+            The coefficient of ``x**deg`` as a polynomial in the same ring.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> R, x, y, z = ring("x, y, z", ZZ)
+        >>> p = 2*x**4 + 3*y**4 + 10*z**2 + 10*x*z**2
+        >>> deg = 2
+        >>> p.coeff_wrt(2, deg) # Using the generator index
+        10*x + 10
+        >>> p.coeff_wrt(z, deg) # Using the generator
+        10*x + 10
+        >>> p.coeff(z**2) # shows the difference between coeff and coeff_wrt
+        10
+
+        See Also
+        ========
+
+        coeff
+        coeffs
+
+        """
+        p = self
+        i = p.ring.index(x)
+        terms = [(m, c) for m, c in p.iterterms() if m[i] == deg]
+
+        if not terms:
+            return p.ring.zero
+
+        monoms, coeffs = zip(*terms)
+        monoms = [m[:i] + (0,) + m[i + 1:] for m in monoms]
+        return p.ring.from_dict(dict(zip(monoms, coeffs)))
+
+    def prem(self, g, x=None):
+        """
+        Computes the pseudo-remainder of the polynomial ``self`` with respect to ``g``.
+
+        Euclidean division is a well-defined operation for univariate polynomials
+        `f` and `g` with coefficients in a field. It yields a unique pair of polynomials
+        `q` and `r`, the quotient and remainder, respectively, satisfying `f = gq + r`,
+        where `deg(r) < deg(g)`.
+
+        For polynomials `f` and `g` with coefficients in a domain, polynomial
+        pseudo-division can be used. It also yields a unique pair of polynomials
+        `q` and `r`, satisfying `mf = gq + r`, where deg(r) < deg(g), `m` is a power
+        of the leading coefficient of `g`, and `f` is pre-multiplied by `m` to ensure
+        exact coefficient divisions.
+
+        In the context of the `prem` method, multivariate polynomials in a ring
+        like `R[x,y,z]` are treated as univariate polynomials with coefficients
+        that are polynomials, such as `R[x,y][z]`. The pseudo-quotient `q` and
+        pseudo-remainder `r` with respect to `z` when dividing `f` by `g`
+        satisfy `mf = gq + r`, where `deg(r,z) < deg(g,z)` and
+        `m = LC(g,z)**(deg(f,z) - deg(g,z)+1)`.
```

Comment:
I didn't really mean for all of this to be included in the docstring. This information belongs somewhere. Perhaps it should be in the `pdiv` docstring and then the other methods should refer to that docstring.

Ideally we would have a shorter explanation though.